src/pages/index.js

```diff
@@ -2,86 +2,166 @@
+  cardsListSection.addItem(createCard(item), firstRender);
+}, cardsListSelector);
 
 const profilePopup = new PopupWithForm(
   "#profile-modal",
   (inputFieldValues, evt) => {
     evt.preventDefault();
-    userInfo.setUserInfo(inputFieldValues);
+    api
+      .setUserInfo(inputFieldValues)
+      .then((res) => {
+        userInfo.setUserInfo(res);
+      })
+      .catch((err) => alert(err))
+      .finally(() => profilePopup.closeAfterSubmit());
   },
   profileFormValidator
 );
 profilePopup.setEventListeners();
 
-const addImagePopup = new PopupWithForm(
+const avatarEditPopup = new PopupWithForm(
+  "#avatar-edit-modal",
+  (inputFieldValue, evt) => {
+    evt.preventDefault();
+    api
+      .setUserAvatar(inputFieldValue.Link)
+      .then((res) => {
+        userInfo.setUserInfo(res);
+      })
+      .catch((err) => alert(err))
+      .finally(() => avatarEditPopup.closeAfterSubmit());
+  },
+  avatarEditFormValidator
+);
+avatarEditPopup.setEventListeners();
+
+const addCardPopup = new PopupWithForm(
   "#add-modal",
   (inputFieldValues, evt) => {
     evt.preventDefault();
-    const { Link: link, Title: title } = inputFieldValues;
-    createCard({ name: title, link: link });
+    api
+      .uploadCard(inputFieldValues)
+      .then((res) => {
+        cardsListSection.addItem(
+          createCard({ name: inputFieldValues.Title, link: inputFieldValues.Link, _id: res._id }),
+          false
+        );
+      })
+      .catch((err) => alert(err))
+      .finally(() => addCardPopup.closeAfterSubmit());
   },
   addFormValidator
 );
-addImagePopup.setEventListeners();
+addCardPopup.setEventListeners();
+
+api
+  .getInitialCards()
+  .then((res) => cardsListSection.renderItems(res, true))
+  .catch((err) => alert(err));
+
+api
+  .getUserInfo()
+  .then((res) => userInfo.setUserInfo(res))
+  .catch((err) => alert(err));
 
 const imagePopup = new PopupWithImage("#image-modal");
 imagePopup.setEventListeners();
 
-const cardsList = new Section(
-  {
-    data: initialCards,
-    renderer: (item) => {
-      createCard(item);
-    },
-  },
-  cardsListSection
-);
-cardsList.renderItems();
+const confirmPopup = new PopupConfirm("#confirm-modal", (cardData, evt) => {
+  evt.preventDefault();
+  api
+    .deleteCard(cardData.cardId)
+    .then(() => {
+      cardData.cardEl.remove();
+      cardData.cardEl = null;
+    })
+    .catch((err) => alert(err));
+});
+confirmPopup.setEventListeners();
 
 //add event listeners
-profileEditButton.addEventListener("click", openProfileForm);
-profileAddButton.addEventListener("click", openAddForm);
-
-//event listener callbacks
-function openProfileForm() {
+profileEditButton.addEventListener("click", () => {
   profilePopup.open();
-  const { name, title } = userInfo.getUserInfo();
-  profileModalNameInput.value = name;
-  profileModalDescInput.value = title;
+  profileModalNameInput.value = userInfo.getUserInfo().name;
+  profileModalDescInput.value = userInfo.getUserInfo().about;
```

Comment:
no need to call `userInfo.getUserInfo()` 2 times. Last time you had this code here

```js

  const { name, about} = userInfo.getUserInfo();
  profileModalNameInput.value = name;
  profileModalDescInput.value = about;
```